homeassistant/components/shelly/utils.py

```diff
@@ -544,15 +545,15 @@ def get_virtual_component_ids(config: dict[str, Any], platform: str) -> list[str
 
 
 @callback
-def async_remove_orphaned_virtual_entities(
+def async_remove_orphaned_entities(
     hass: HomeAssistant,
     config_entry_id: str,
     mac: str,
     platform: str,
-    virt_comp_type: str,
-    virt_comp_ids: list[str],
+    keys: list[str],
```

Comment:
```suggestion
    keys: Iterable[str],
```

I think this only needs to be an `Iterable`, than you can avoid making copies